src/audio/kpb.c

```diff
@@ -171,45 +179,52 @@ static int kpb_prepare(struct comp_dev *dev)
 
 	cd->no_of_clients = 0;
 
-	/* allocate history_buffer/s */
-#if KPB_NO_OF_HISTORY_BUFFERS == 2
+	/* allocate history buffer/s */
+	while (hb_size > 0 && i < ARRAY_SIZE(hb_mcp)) {
+		ptr = rballoc(RZONE_RUNTIME, hb_mcp[i], ca_size);
+
+		if (ptr) {
+			history_buffer->sta_addr = ptr;
+			history_buffer->end_addr = ptr + ca_size;
+			history_buffer->w_ptr = ptr;
+			history_buffer->r_ptr = ptr;
+			history_buffer->state = KPB_BUFFER_FREE;
+			hb_size = hb_size - ca_size;
+			history_buffer->next = &cd->history_buffer;
+
+			/* Do we need another buffer? */
+			if (hb_size > 0) {
+				new_hb = rzalloc(RZONE_RUNTIME,
```

Comment:
What's the difference between this allocation and the one at line 184. Can we comment what we are allocating for each.

Reply:
Sure, I will put more comments. The difference between these two allocation is one allocates the structure for new history buffer and the other allocates the actual data where we store audio.